frappe/core/doctype/submission_queue/submission_queue.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) 2022, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from rq import get_current_job
+from rq.job import Job
+from rq.registry import FailedJobRegistry
+
+import frappe
+from frappe import _
+from frappe.desk.doctype.notification_log.notification_log import enqueue_create_notification
+from frappe.model.document import Document
+from frappe.monitor import add_data_to_monitor
+from frappe.utils import now, time_diff_in_seconds
+from frappe.utils.background_jobs import get_queue, get_redis_conn
+from frappe.utils.data import cint
+
+
+class SubmissionQueue(Document):
+	@property
+	def created_at(self):
+		return self.creation
+
+	@property
+	def enqueued_by(self):
+		return self.owner
+
+	@property
+	def queued_doc(self):
+		return getattr(self, "to_be_queued_doc", frappe.get_doc(self.ref_doctype, self.ref_docname))
+
+	@staticmethod
+	def clear_old_logs(days=30):
+		from frappe.query_builder import Interval
+		from frappe.query_builder.functions import Now
+
+		table = frappe.qb.DocType("Submission Queue")
+		frappe.db.delete(table, filters=(table.modified < (Now() - Interval(days=days))))
+
+	def insert(self, to_be_queued_doc: Document, action: str):
+		self.to_be_queued_doc = to_be_queued_doc
+		self.action_for_queuing = action
+		super().insert(ignore_permissions=True)
+
+	def lock(self):
+		self.queued_doc.lock()
+
+	def unlock(self):
+		self.queued_doc.unlock()
+
+	def after_insert(self):
+		self.queue_action(
+			"background_submission",
+			to_be_queued_doc=self.queued_doc,
+			action_for_queuing=self.action_for_queuing,
+			timeout=600,
+			enqueue_after_commit=True,
+		)
+
+	def background_submission(self, to_be_queued_doc: Document, action_for_queuing: str):
+		current_job = get_current_job()
+
+		# Set the job id for that submission doctype
+		frappe.db.set_value(
+			self.doctype,
+			self.name,
+			{"job_id": current_job.id},
+			update_modified=False,
+		)
```

Comment:
add a commit after this - rather take out the set_value and commit to a separate method ..something like `update_job_id`

Reply:
done 👍🏿